GitCommands/UserRepositoryHistory/RemoteRepositoryManager.cs

```diff
@@ -0,0 +1,130 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+
+namespace GitCommands.UserRepositoryHistory
+{
+    /// <summary>
+    /// Manages the history of remote git repositories.
+    /// </summary>
+    public sealed class RemoteRepositoryManager : IRepositoryManager
```

Comment:
❓ How much of this file is duplicated from above? Can we use a shared based class where functionality is the same?

Reply:
As per the PR description - there are very similar, however there are nuanced differences.
I have thought of a base class, however I am very much against the inheritance because way to often it gets misused and leads to horrible incoherent implementations.
I don't like duplications almost to the same extent as I don't like the inheritance, however there is a threshold to which I can tolerate it - e.g. [the rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)).

Having said that, I am still revisit these API.

Reply:
📝 I'm interested in taking this discussion further, but it's not the focus of my initial review.